content/client-lib-development-guide/features.textile

```diff
@@ -1361,15 +1370,42 @@ h4. ChannelOccupancy
 
 h4. ChannelMetrics
 
-* @(CHM1)@ @ChannelMetrics@ is a type that contains the count of publishers and subscribers, connections and presenceConnections, presenceMembers and presenceSubscribers 
+* @(CHM1)@ @ChannelMetrics@ is a type that contains the count of publishers and subscribers, connections and presenceConnections, presenceMembers and presenceSubscribers
 * @(CHM2)@ The attributes of @ChannelMetrics@ consist of:
 ** @(CHM2a)@ @connections@ integer - the total number of connections to the channel
 ** @(CHM2b)@ @presenceConnections@ integer - the total number of presence connections to the channel
 ** @(CHM2c)@ @presenceMembers@ integer - the total number of presence members for the channel
 ** @(CHM2d)@ @presenceSubscribers@ integer - the total number of presence subscribers for the channel
-** @(CHM2e)@ @publishers@ integer - the total number of publishers to the channel 
+** @(CHM2e)@ @publishers@ integer - the total number of publishers to the channel
 ** @(CHM2f)@ @subscribers@ integer - the total number of subscribers to the channel
 
+h4. BatchResult
+* @(BPA1)@ Contains the results from the batch operation
+* @(BPA2)@ @BatchResult@ has the following attributes:
+** @(BPA2a)@ @responses@ is an array of batch response objects. undefined if the request failed completely (e.g. for an invalid key)
+** @(BPA2b)@ @error@ is an @ErrorInfo@ object which is populated if one or more batch publish requests failed
+
+h4. BatchPublishResponse
+* @(BPB1)@ Contains information for each batch publish request within a @BatchResult@
+* @(BPB2)@ @BatchPublishResponse@ has the following attributes:
+** (@BPB2a)@ @channel@ is the channel name which this publish request was directed to
+** (@BPB2b)@ @messageId@ contains the resultant message ID, if the request succeeds and is undefined if @error@ is present
+** (@BPB2c)@ @error@ contains an @ErrorInfo@ object if this publish request failed, and is undefined if it succeeded
```

Comment:
Same comment again as line 1384
Idiomatic Go would return a pointer to a result (nil if no result) and and error.
Like result, err := REST.batch.publish()
So I'm not sure if we should be prescribing that BatchPublishResult has an error nested inside it. Maybe we can leave this open to being implemented idiomatically by each language.